src/components/Icon/Icon.tsx

```diff
@@ -19,8 +20,30 @@ const COLORS_WITH_BACKDROPS = [
 // styleguide to generate the props explorer
 interface Props extends IconProps {}
 
+function parseSource(source: string | Function) {
+  if (typeof source === 'function') {
+    return source.name;
+  } else if (source === 'placeholder') {
+    return source;
+  }
+  return 'custom icon string';
+}
+
 export function Icon({source, color, backdrop, accessibilityLabel}: Props) {
   const i18n = useI18n();
+  const telemetry = useTelemetry();
+
+  /* eslint-disable babel/camelcase */
+  useEffect(() => {
+    telemetry.produce('polaris_icons_usage/1.0', {
+      icon_source: parseSource(source),
+      color,
+      backdrop,
+      accessibility_label: accessibilityLabel,
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I think we should keep the rules of hooks happy. We should either:

- Track color/backdrop/accessibilityLabel which includes when they change
- Don't track them at all

Currently we're in an odd halfway house where we track initial values but if the color ever changes then we don't track it which feels like we're building an incomplete dataset.

My gut is saying that tracking these will make it harder to sort through the data as it makes it harder to reason about "what icons were loaded" as we have to discount any cases where just the color/backdrop/label were changed